docs/who-uses-meson-python.rst

```diff
@@ -0,0 +1,26 @@
+.. _who-uses-meson-python:
+
+**************************
+Who uses ``meson-python``?
+**************************
```

Comment:
I don't like adding a list of projects using meson-python to the documentation, or anywhere else. As we don't have any need for advertisement (and I'm pretty sure we don't want to be advertising any other project) it does not serve any purpose and it is going to be a burden to maintain. For the users, it is much better to have a short and curated list of projects that follow best practices and that can be used as examples.

Unless the listed projects are known to following best practices, the list is not going to be useful for the users. I don't think we have resources to review all the projects submitted to be listed, and to keep our review current.

I would much prefer it this would be something like:
```
 Curated list of projects using ``meson-python``
===================================

The following projects use Meson as their build system and meson-python to produce Python wheels. They are roughly listed in order of complexity:

- `siphash24 <https://github.com/dnicolodi/siphash24>`_ A very simple project. It demonstrates how Meson makes it trivial to compile a CPython extension written in Cython via a simple tempate engine and link it to a library compiled from a Meson subproject. Also an example of how to use `cibuildwheel`_ to produce Python wheels for several platforms.

- (i don't know the peculiarities of the other projects)

- `SciPy <https://github.com/scipy/scipy>`_ Probably the most complex Python project using Meson and ``meson-python``. Staring at the build definition too long may cause headaches. 

```

Reply:
Are the new updates good for you, or do you change the wording further?

Reply:
@dnicolodi please let me know if you'd like further changes, I think this is the only blocker for merging this.

Reply:
I'm not completely happy with the section name. I would prefer something like "Projects using meson-python".